bluebutton-data-pipeline-app/src/test/java/gov/hhs/cms/bluebutton/datapipeline/app/S3ToFhirLoadAppIT.java

```diff
@@ -0,0 +1,425 @@
+	 *            of
+	 */
+	private static void verifyExitValueMatchesSignal(int signalNumber, Process process) {
+		/*
+		 * Per POSIX (by way of http://unix.stackexchange.com/a/99143),
+		 * applications that exit due to a signal should return an exit code
+		 * that is 128 + the signal number.
+		 */
+		Assert.assertEquals(128 + signalNumber, process.exitValue());
+	}
+
+	/**
+	 * Ensures that {@link FhirTestUtilities#cleanFhirServer()} is called after
+	 * each test case.
+	 */
+	@After
+	public void cleanFhirServerAfterEachTestCase() {
+		FhirTestUtilities.cleanFhirServer(FHIR_API);
+	}
+
+	/**
+	 * @param bucket
+	 *            the S3 {@link Bucket} that the application will be configured
+	 *            to pull RIF data from
+	 * @return a {@link ProcessBuilder} that can be used to launch the
+	 *         application
+	 */
+	private static ProcessBuilder createAppProcessBuilder(Bucket bucket) {
+		String[] command = createCommandForCapsule();
+		ProcessBuilder appRunBuilder = new ProcessBuilder(command);
+
+		appRunBuilder.environment().put(AppConfiguration.ENV_VAR_KEY_BUCKET, bucket.getName());
+		appRunBuilder.environment().put(AppConfiguration.ENV_VAR_KEY_FHIR, FHIR_API);
+		/*
+		 * Note: Not explicitly providing AWS credentials here, as the child
+		 * process will inherit any that are present in this build/test process.
+		 */
+		return appRunBuilder;
+	}
+
+	/**
+	 * @return the command array for
+	 *         {@link ProcessBuilder#ProcessBuilder(String...)} that will launch
+	 *         the application via its <code>.x</code> capsule executable
+	 */
+	private static String[] createCommandForCapsule() {
+		try {
+			Path buildTargetDir = Paths.get(".", "target");
+			Path appExe = Files.list(buildTargetDir)
+					.filter(f -> f.getFileName().toString().startsWith("bluebutton-data-pipeline-app-"))
+					.filter(f -> f.getFileName().toString().endsWith("-capsule-fat.x")).findFirst().get();
+
+			return new String[] { appExe.toAbsolutePath().toString() };
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Managing external processes is tricky: at the OS level, all processes'
+	 * output is sent to a buffer. If that buffer fills up (because you're not
+	 * reading the output), the process will block -- forever. To avoid that,
+	 * it's best to always have a separate thread running that consumes a
+	 * process' output. This {@link ProcessOutputConsumer} is designed to allow
+	 * for just that.
+	 */
+	private static final class ProcessOutputConsumer implements Runnable {
+		private final BufferedReader stdoutReader;
+		private final StringBuffer stdoutContents;
+
+		/**
+		 * Constructs a new {@link ProcessOutputConsumer} instance.
+		 * 
+		 * @param the
+		 *            {@link ProcessOutputConsumer} whose output should be
+		 *            consumed
+		 */
+		public ProcessOutputConsumer(Process process) {
+			/*
+			 * Note: we're only grabbing STDOUT, because we're assuming that
```

Comment:
It doesn't seem conventional to be logging to STDOUT...is this just because we are running it as an external process and that is the only way to get feedback on what the process is doing?